src/helm/proxy/clients/huggingface_client.py

```diff
@@ -236,9 +236,16 @@ def do_it():
                     ]:
                         tokens = [tokenizer.convert_tokens_to_string([i]) for i in tokenizer.tokenize(request.text)]
                     else:
-                        tokens = tokenizer.tokenize(request.text)
-                        # TODO(1522): Reenable this to revove "▁"
+                        # Tokenizes the text and returns the tokens as a list of strings,
```

Comment:
Ah so it sounds like tokenizers that don't contain `gpt` or are not one of the listed models above handle the `_` character by turning it into an empty string -- are we sure that every model other than those above has this problem? Or should we add a special case for the model names which we know we must use `cleanup_tokens`? For example, some random model might tokenize `"Hello world"` as `["Hello", "@world"]` or something, in which case `cleanup_tokens` would have to be edited to also replace `@` with a space. In this case, we should probably make `[tokenizer.convert_tokens_to_string([i]) for i in tokenizer.tokenize(request.text)]` the default behavior and add special-cased model names for `cleanup_tokens`.

Reply:
That,s a good point. So right now, I am only aware of this case, but you might be right. What I propose is to add the tokenizer as an argument of `cleanup_tokens()`. Right now it does nothing but if someday e add a model that needs to have its own special case, then we can do it.